src/Raven.Client/Documents/Subscriptions/SubscriptionState.cs

```diff
@@ -16,17 +16,22 @@ namespace Raven.Client.Documents.Subscriptions
     public class SubscriptionState : IDatabaseTask, IDatabaseTaskStatus
     {
         public string Query { get; set; }
-        public string ChangeVectorForNextBatchStartingPoint { get; set; }
-        public Dictionary<string, string> ChangeVectorForNextBatchStartingPointPerShard { get; set; }
         public long SubscriptionId { get; set; }
         public string SubscriptionName { get; set; }
         public string MentorNode { get; set; }
-        public string NodeTag { get; set; }
         public DateTime? LastBatchAckTime { get; set; }  // Last time server made some progress with the subscriptions docs  
         public DateTime? LastClientConnectionTime { get; set; } // Last time any client has connected to server (connection dead or alive)
-        
         public bool Disabled { get; set; }
 
+        // for non-sharded
+        public string NodeTag { get; set; }
+        public string ChangeVectorForNextBatchStartingPoint { get; set; }
+
+        // for sharded
+        public Dictionary<string, string> ChangeVectorForNextBatchStartingPointPerShard { get; set; }
+        public Dictionary<string, string> NodeTagPerShard { get; set; }
+        public Dictionary<long, string> IgnoreBucketLesserChangeVector { get; set; }
+
```

Comment:
Can a 5.x client work with 6.x sharded subscription?

Reply:
Let's change this so we have _one_ dictionary, with all the fields in there.

Reply:
We definitely will check that old client can work with the new _sharded_ server, but I don't see a reason why not.
Changed that to be a single class called `SubscriptionShardingState`